src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${fromSeparator}`);
+  let year = '';
+  let month = '';
+  let day = '';
+  const result = [];
+
+  for (let i = 0; i <= dateParts.length; i++) {
```

Comment:
The loop condition should be 'i < dateParts.length' instead of 'i <= dateParts.length' to avoid an off-by-one error that would cause an undefined value to be accessed.